st2actions/st2actions/runners/ssh/paramiko_ssh.py

```diff
@@ -349,6 +349,7 @@ def run(self, cmd, timeout=None, quote=False):
         chan = transport.open_session()
 
         start_time = time.time()
+        chan.get_pty()
```

Comment:
https://github.com/fabric/fabric/blob/master/fabric/operations.py#L721 Fabric always sets pty=True unless you explicitly disable it. This might explain why puppet_bootstrap always succeeds with fabric.